beacon_node/network/src/sync/manager.rs

```diff
@@ -209,6 +213,36 @@ pub enum BatchProcessResult {
     NonFaultyFailure,
 }
 
+/// The result of processing multiple data columns.
+#[derive(Debug)]
+pub enum CustodyBatchProcessResult {
+    /// The custody batch was completed successfully. It carries whether the sent batch contained data columns.
+    Success {
+        batch_id: Epoch,
+        #[allow(dead_code)]
+        sent_columns: usize,
+        imported_columns: usize,
+    },
+    /// The custody batch processing failed. It carries whether the processing imported any data columns.
+    FaultyFailure {
+        batch_id: Epoch,
+        penalty: PeerAction,
+    },
+    NonFaultyFailure {
+        batch_id: Epoch,
+    },
```

Comment:
Can be simplified into
```suggestion
    Failure {
        batch_id: Epoch,
        penalty: Option<PeerAction>,
    },
```

Reply:
Or even

```rs
CustodyBatchProcessed { 
  batch_id: Epoch,
  result: Result<(), CustodyBatchProcessError>
}

CustodyBatchProcessError {
  penalty: Option<PeerAction>,
}
```

`imported_columns` is only used for a log that not super relevant and can be moved into the importer function

Reply:
Removed the extra enum, now we only have a Success and failure enum, w/ an optional peer action penalty in the failure case